### DISCUSSION TITLE
rbenv install 3.0.2 fails on macOS Big Sur 11.6 (20G165)

### DISCUSSION THREAD

**Discussion by phirk** [rbenv/ruby-build#1866] [Build failures]
`rbenv install 3.0.2` fails on macOS Big Sur 11.6 (20G165):

```
➜ rbenv install 3.0.2
Downloading ruby-3.0.2.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.2.tar.gz
Installing ruby-3.0.2...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.6 using ruby-build 20210925)

Inspect or clean up the working tree at /var/folders/n5/99pd__cd4hldsmzrcr6zyr600000gp/T/ruby-build.20210929113846.45344.8kAFvG
Results logged to /var/folders/n5/99pd__cd4hldsmzrcr6zyr600000gp/T/ruby-build.20210929113846.45344.log

Last 10 log lines:
The Ruby yaml extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/rob/.rbenv/versions/3.0.2
  --enable-shared
  --with-readline-dir=/usr/local/opt/readline
  --with-openssl-dir=/usr/local/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/rob/.rbenv/versions/3.0.2/lib
  CPPFLAGS=-I/Users/rob/.rbenv/versions/3.0.2/include
```

Full log: https://gist.github.com/phirk/c0e6920602f6a3716de4cb99c30e6adc

**Comment by phirk** [comment]
Note: these (required) libraries are all installed by Homebrew:

* openssl@1.1
* readline
* libyaml

**Comment by soffes** [comment]
I was having a similar issue because I accidentally installed `openssl@3`. This fixed it for me:

```
$ brew install openssl@1.1
$ RUBY_CONFIGURE_OPTS="--with-openssl-dir=`brew --prefix openssl@1.1`" rbenv install 3.0.2
```

(If this doesn't work, try getting the value of `brew --prefix openssl@1.1` and using that directly instead.)

**Comment by minhtienvu** [comment]
I have the same issue when I try to install 3.0.2 verssion.
```
RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rben
v install 3.0.2
Downloading ruby-3.0.2.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.2.tar.gz
Installing ruby-3.0.2...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.6 using ruby-build 20210928)

Inspect or clean up the working tree at /var/folders/kg/jgp0sjw16pg5j3lmp9yq9_140000gn/T/ruby-build.20211018170536.62572.B2UNRu
Results logged to /var/folders/kg/jgp0sjw16pg5j3lmp9yq9_140000gn/T/ruby-build.20211018170536.62572.log

Last 10 log lines:
checking for dtrace... dtrace
checking for dot... no
checking for doxygen... no
checking for pkg-config... pkg-config
checking whether it is Android... no
checking for cd using physical directory... cd -P
checking whether CFLAGS is valid... yes
checking whether LDFLAGS is valid... no
configure: error: something wrong with LDFLAGS="-L/Users/vuminhtien/.rbenv/versions/3.0.2/lib "
make: *** No targets specified and no makefile found.  Stop.
```

**Comment by briancolfer** [comment]
What worked for me ... first uninstall the existing 3.0.2 then reinstall with the above options.

**Comment by shelbykohring** [comment]
> I was having a similar issue because I accidentally installed `openssl@3`. This fixed it for me:
> 
> ```
> $ brew install openssl@1.1
> $ RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rbenv install 3.0.2
> ```
> 
> Note: I'm on an M1. You'll need to change the path from `/opt` to `/usr/local` if you are using Intel.

I am running into this issue on Mac OS 12.0.1 Monterey using Intel. Unsure with '/opt' path should be changed to 'usr/local' . 

Have tried changing the first one, then tried the second one, then tried them both. Nothing is resolving this issue for me. Anyone have success installing Ruby 3.0.2 on Monterey?

**Comment by soffes** [comment]
@shelbykohring updated [my comment](https://github.com/rbenv/ruby-build/issues/1801#issuecomment-936520879). Hope that helps!

**Comment by edmundask** [comment]
> I have the same issue when I try to install 3.0.2 verssion.
> 
> ```
> RUBY_CONFIGURE_OPTS="--with-openssl-dir=/opt/homebrew/opt/openssl@1.1" rben
> v install 3.0.2
> Downloading ruby-3.0.2.tar.gz...
> -> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.2.tar.gz
> Installing ruby-3.0.2...
> ruby-build: using readline from homebrew
> 
> BUILD FAILED (macOS 11.6 using ruby-build 20210928)
> 
> Inspect or clean up the working tree at /var/folders/kg/jgp0sjw16pg5j3lmp9yq9_140000gn/T/ruby-build.20211018170536.62572.B2UNRu
> Results logged to /var/folders/kg/jgp0sjw16pg5j3lmp9yq9_140000gn/T/ruby-build.20211018170536.62572.log
> 
> Last 10 log lines:
> checking for dtrace... dtrace
> checking for dot... no
> checking for doxygen... no
> checking for pkg-config... pkg-config
> checking whether it is Android... no
> checking for cd using physical directory... cd -P
> checking whether CFLAGS is valid... yes
> checking whether LDFLAGS is valid... no
> configure: error: something wrong with LDFLAGS="-L/Users/vuminhtien/.rbenv/versions/3.0.2/lib "
> make: *** No targets specified and no makefile found.  Stop.
> ```

@minhtienvu the problem for me was the following line:

```sh
export LIBRARY_PATH="${LD_LIBRARY_PATH:+LD_LIBRARY_PATH:}/usr/local/opt/openssl/lib/"
```

I have `openssl` installed via homebrew, so commenting this line out inside `.zshrc`, `.bash_profile`, `.zprofile` may help. Note that this line is somewhat custom on my setup, but you may have something similar. I'm running `Monterey 12.0.1` on the new MacBook Pro 16" M1, btw.

**Reply by gotoAndBliss** [reply]
Thanks so much for writing this. 

On an M1, recent upgrade with an old `.zshrc` file.

Like you, commenting this out did the trick. 

```
# export LIBRARY_PATH=$LIBRARY_PATH:/usr/local/lib:/opt/homebrew/lib/
``` 

I'm assuming the extended paths were overriding what Homebrew was capable of handling alone?

**Comment by a-sh-dev** [comment]
> `rbenv install 3.0.2` fails on macOS Big Sur 11.6 (20G165):
> 
> ```
> ➜ rbenv install 3.0.2
> Downloading ruby-3.0.2.tar.gz...
> -> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.2.tar.gz
> Installing ruby-3.0.2...
> ruby-build: using readline from homebrew
> 
> BUILD FAILED (macOS 11.6 using ruby-build 20210925)
> 
> Inspect or clean up the working tree at /var/folders/n5/99pd__cd4hldsmzrcr6zyr600000gp/T/ruby-build.20210929113846.45344.8kAFvG
> Results logged to /var/folders/n5/99pd__cd4hldsmzrcr6zyr600000gp/T/ruby-build.20210929113846.45344.log
> 
> Last 10 log lines:
> The Ruby yaml extension was not compiled.
> ERROR: Ruby install aborted due to missing extensions
> Configure options used:
>   --prefix=/Users/rob/.rbenv/versions/3.0.2
>   --enable-shared
>   --with-readline-dir=/usr/local/opt/readline
>   --with-openssl-dir=/usr/local/opt/openssl@1.1
>   CC=clang
>   LDFLAGS=-L/Users/rob/.rbenv/versions/3.0.2/lib
>   CPPFLAGS=-I/Users/rob/.rbenv/versions/3.0.2/include
> ```
> 
> Full log: https://gist.github.com/phirk/c0e6920602f6a3716de4cb99c30e6adc

I have the same exact issue with the same error message.
Tried the normal `rbenv install 3.0.2`, but was unsuccessful. 
Tried `CFLAGS="-Wno-error=implicit-function-declaration" rbenv install 3.0.2` based on [this article](https://dev.to/rbazinet/fix-installation-of-ruby-using-rbenv-on-macos-big-sur-3432). 

I also have the `openssl@1.1` installed and  running `macOS Big Sur 11.2.3`.

Tried running the suggested 
```
RUBY_CONFIGURE_OPTS="--with-openssl-dir=`brew --prefix openssl@1.1`" rbenv install 3.0.2
```
but unfortunately still unsuccessful and I got the same error as OP.

**Comment by a-sh-dev** [comment]
So, I updated my BigSur to 11.6.1 (20G224), which includes `xcode CLT 13` also installed through the software update) until no more update was available in the list except for the macOS Monterey upgrade option.

Restarted my MacBook, and tried to run ` rbenv install 3.0.2`, and it somehow was installed successfully... 

I'm not sure if it's part of the reason, but prior to that, I ran `brew doctor` and `brew cleanup` and found issues with my `node` not brew linked due to non-brew installation previously... although this node linking issue hasn't been resolved yet, the rbenv ruby installation on version 3.0.2 was successful.

**Comment by aeldaly-tc** [comment]
I've tried everything suggested here, but nothing seems to work.

Here's my latest attempt:

```
 ↵ 1  LDFLAGS="-L/Users/aelday/.rbenv/versions/3.0.3/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/python@3.8/lib" CPPFLAGS="-I/Users/aelday/.rbenv/versions/3.0.3/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/libffi/include" RUBY_CONFIGURE_OPTS="--with-openssl-dir=/usr/local/opt/openssl@1.1" rbenv install
Downloading ruby-3.0.3.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.3.tar.gz
Installing ruby-3.0.3...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.6.2 using ruby-build 20220125)

Inspect or clean up the working tree at /var/folders/z9/htgc96px1gb66fjrkyht06bw0000gq/T/ruby-build.20220201101206.59614.Ys9mtP
Results logged to /var/folders/z9/htgc96px1gb66fjrkyht06bw0000gq/T/ruby-build.20220201101206.59614.log

Last 10 log lines:
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/aelday/.rbenv/versions/3.0.3
  --enable-shared
  --with-readline-dir=/usr/local/opt/readline
  --with-openssl-dir=/usr/local/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/aelday/.rbenv/versions/3.0.3/lib -L/Users/aelday/.rbenv/versions/3.0.3/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl@1.1/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/python@3.8/lib
  CPPFLAGS=-I/Users/aelday/.rbenv/versions/3.0.3/include -I/Users/aelday/.rbenv/versions/3.0.3/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@1.1/include -I/usr/local/opt/libffi/include
zsh: exit 1
```

macOS Big Sur - 11.6.2

**Reply by aeldaly-tc** [reply]
Also tried this:

```
 ↵ 1  RUBY_CONFIGURE_OPTS="--with-openssl-dir=`brew --prefix openssl@1.1`" rbenv install                               35% (...) 🔋  10.39 L
Downloading ruby-3.0.3.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.0/ruby-3.0.3.tar.gz
Installing ruby-3.0.3...
ruby-build: using readline from homebrew

BUILD FAILED (macOS 11.6.2 using ruby-build 20220125)

Inspect or clean up the working tree at /var/folders/z9/htgc96px1gb66fjrkyht06bw0000gq/T/ruby-build.20220201102315.10589.83glTg
Results logged to /var/folders/z9/htgc96px1gb66fjrkyht06bw0000gq/T/ruby-build.20220201102315.10589.log

Last 10 log lines:
The Ruby openssl extension was not compiled.
ERROR: Ruby install aborted due to missing extensions
Configure options used:
  --prefix=/Users/aelday/.rbenv/versions/3.0.3
  --enable-shared
  --with-readline-dir=/usr/local/opt/readline
  --with-openssl-dir=/usr/local/opt/openssl@1.1
  CC=clang
  LDFLAGS=-L/Users/aelday/.rbenv/versions/3.0.3/lib -L/usr/local/opt/readline/lib -L/usr/local/opt/openssl@3/lib -L/usr/local/opt/libffi/lib -L/usr/local/opt/python@3.8/lib
  CPPFLAGS=-I/Users/aelday/.rbenv/versions/3.0.3/include -I/usr/local/opt/readline/include -I/usr/local/opt/openssl@3/include -I/usr/local/opt/libffi/include
zsh: exit 1
```

Still no luck

**Reply by chinmayv** [reply]
For me, oh-my-zsh plugin directory had rbenv directory. I deleted that directory and install worked after that.
